src/AbsBackup.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ProjectBackup;
+
+use GuzzleHttp\Client;
+use Keboola\StorageApi\Client as StorageApi;
+use Keboola\StorageApi\HandlerStack;
+use Keboola\StorageApi\Options\GetFileOptions;
+use Keboola\Temp\Temp;
+use MicrosoftAzure\Storage\Blob\BlobRestProxy;
+use Psr\Log\LoggerInterface;
+
+class AbsBackup extends Backup
+{
+    private BlobRestProxy $absClient;
+
+    private string $path;
+
+    public function __construct(
+        StorageApi $sapiClient,
+        BlobRestProxy $absClient,
+        string $path,
+        ?LoggerInterface $logger = null
+    ) {
+        $this->absClient = $absClient;
+        $this->path = $path;
+
+        parent::__construct($sapiClient, $logger);
+    }
+
+    protected function putToStorage(string $name, string $content): void
+    {
+        $this->absClient->createBlockBlob(
+            $this->path,
+            $name,
+            $content
+        );
+    }
+
+    public function backupTable(string $tableId): void
+    {
+        $table = $this->sapiClient->getTable($tableId);
+
+        if ($table['bucket']['stage'] === 'sys') {
+            $this->logger->warning(sprintf('Skipping table %s (sys bucket)', $table['id']));
+            return;
+        }
+
+        if ($table['isAlias']) {
+            $this->logger->warning(sprintf('Skipping table %s (alias)', $table['id']));
+            return;
+        }
+
+        $this->logger->info(sprintf('Exporting table %s', $tableId));
+
+        $tmp = new Temp();
+        $tmp->initRunFolder();
+
+        $fileId = $this->sapiClient->exportTableAsync($tableId, [
+            'gzip' => true,
+        ]);
+        $fileInfo = $this->sapiClient->getFile($fileId['file']['id'], (new GetFileOptions())->setFederationToken(true));
+
+        // Initialize BlobRestProxy with credentials from Storage API
+        $absClient = BlobRestProxy::createBlobService($fileInfo['absCredentials']['SASConnectionString']);
+
+        if ($fileInfo['isSliced'] === true) {
+            // Download manifest with all sliced files
+            $client = new Client([
+                'handler' => HandlerStack::create([
+                    'backoffMaxTries' => 10,
+                ]),
+            ]);
+            $manifest = json_decode($client->get($fileInfo['url'])->getBody()->getContents(), true);
```

Comment:
Na stahovanie súborov by si mal IMHO pouzit Storage API klienta, ktory ma vyriesenu podporu pre S3 a ABS:

https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/Downloader/DownloaderInterface.php

metoda v klientovi sa vola `downloadFile`: https://github.com/keboola/storage-api-php-client/blob/ecf563c458596ce7ef917e31899208e5b5d1b540/src/Keboola/StorageApi/Client.php#L1743

Reply:
Inak v klientovi su aj triedy `AbsUploader` a `S3Uploader`, ktore vedia bez namahy nahravat subor paralelne ... staci im dat do konstruktora spojenie. Podla mna by tiez stalo za to ich pouzit, ako to vymyslat znova:

https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/ABSUploader.php
https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/S3Uploader.php

Reply:
... a vlastne ani nemusis stahovat jednotlive subory, ale mas tam rovno `TableExporter`: https://github.com/keboola/storage-api-php-client/blob/master/src/Keboola/StorageApi/TableExporter.php ... ktory to ma vsetko vyriesene pre S3 aj ABS, ... potrebujes iba vytvorit spojenie, ... a potom stiahnuty súbor nahrat do cielovej S3/ABS.

Uplne idealne by bolo kopirovat subory priamo keboola S3/ABS -> backup S3/ABS .(bez stahovania).. ale to neviem ci by slo, a urcite to teraz neriesme.

Reply:
to co tu popisuješ je spíš na refactoring celé metody, což imho není předmětem tohoto PR

souhlasím že by bylo fajn to udělat jak popisuješ, ale má to další dopady na obnovování projektu (změnila by se struktura v zálohách se kterou by bylo třeba počítat) atd... a na to by bylo potřeba více času než jsem tomu věnoval a nechal bych to spíše na další sprinty

tady jsem jen přidal podporu zálohování na ABS a princip jak se to zálohuje jsem se snažil nechat stejný jaký byl